runtime-sdk/modules/evm/src/lib.rs

```diff
@@ -54,6 +55,9 @@ pub mod state {
 pub trait Config: 'static {
     /// Module that is used for accessing accounts.
     type Accounts: modules::accounts::API;
+
+    /// Ethereum-format transactions must use this chain ID.
+    const EXPECTED_CHAIN_ID: Option<u64>;
```

Comment:
I would change this to just `CHAIN_ID`, make it non optional and update the evm backend to also return this when querying for `chain_id` instead of using that derivation scheme.

Reply:
the rename is fine. as you observe below, none refers to legacy transactions with no chain separation, not "accept anything"

Reply:
I don't think we want to support transactions without chain separation though? In any case the same `CHAIN_ID` should also be used in the `evm_backend.rs` for chain ID queries from EVM.

Reply:
oh ok, that'll make things easier for the other places that'll use it

Reply:
changed

Reply:
Also don't forget to update the chain ID query in the EVM backend.

Reply:
oh yeah, that always sneaks around these refactors :laughing: